docs/src/main/asciidoc/spanner.adoc

```diff
@@ -0,0 +1,129 @@
+== Spring Data
+
+http://projects.spring.io/spring-data/[Spring Data]
+is an abstraction for storing and retrieving POJOs in numerous storage technologies.
+Spring Cloud GCP adds Spring Data support for Google Cloud Spanner.
+
+A Spring Boot starter is provided to auto-configure the various Storage components.
+
+Maven coordinates, using Spring Cloud GCP BOM:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.springframework.cloud</groupId>
+    <artifactId>spring-cloud-gcp-starter-data-spanner</artifactId>
+</dependency>
+----
+
+Gradle coordinates:
+
+[source,subs="normal"]
+----
+dependencies {
+    compile group: 'org.springframework.cloud', name: 'spring-cloud-gcp-starter-data-spanner'
+}
+----
+
+
+=== Google Cloud Spanner
+
+Spring Data Spanner allows you to define entity classes and corresponding repository interfaces:
+
+[source,java]
+----
+@SpannerTable(name = "students")
+public class SimpleStudent {
+	@Id
+	String id;
+
+	@SpannerColumn(name = "first_name")
+	String name;
+
+	@SpannerColumnInnerType(innerType = String.class)
+	List<String> teachers;
+
+	int age;
+}
+----
+
+The `@SpannerTable` annotation provides the name of the Spanner table that stores instances of this class, one per row.
+The `@Id` annotation is required and identifies the property corresponding to the primary key column.
+The `@SpannerColumn` annotation optionally provides a different column name than that of the property.
+The `@SpannerColumnInnerType` annotation is required for List properties due to Java's type erasure.
+
+[source,java]
+----
+@Repository
+public interface StudentRepository extends PagingAndSortingRepository<SimpleStudent, String> {
+
+	List<SimpleStudent> findTop3DistinctByIdAndNameOrAgeOrderByAgeDesc(
+			String id, String name, int age);
+
+	int countByAgeLessThanEqual(int age);
+
+	// This method uses the query from the properties file instead of one generated based on name.
+	List<SimpleStudent> fetchBySomeQuery(String someArg);
+}
+
+----
+
+These can be used in an application like this:
+
+[source,java]
+----
+@EnableSpannerRepositories(namedQueriesLocation = "classpath:/spanner-named-queries.properties")
+public class MyApplication {
+
+	@Autowired
+	SpannerOperations spannerOperations;
+
+	@Autowired
+	StudentRepository studentRepository;
+
+	public void demo() {
+
+	  // storing new students
+	  SimpleStudent ss = new SimpleStudent();
+	  ss.name = "student1";
+	  ss.age = 99;
+	  spannerOperations.insert(ss);
+
+	  // getting students from Spanner
+	  List<SimpleStudent> ssList =
+	    studentRepository.findTop3DistinctByIdAndNameOrAgeOrderByAgeDesc("someId", "bob", 12);
+
+	  List<SimpleStudent> ssList2 = studentRepository.fetchBySomeQuery("some argument");
+	}
+}
+
+----
+
+The https://docs.spring.io/spring-data/jpa/docs/1.5.0.RELEASE/reference/html/repositories.html[query methods]
```

Comment:
use https://docs.spring.io/spring-data/commons/docs/current/reference/html/#repositories.query-methods instead, that's the spring data commons link for query methods.